extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/DeploymentDecorator.java

```diff
@@ -0,0 +1,35 @@
+package io.quarkus.kubernetes.deployment;
+
+import java.util.Optional;
+
+import io.fabric8.kubernetes.client.Config;
+
+public interface DeploymentDecorator<D extends DeploymentDecorator<D>> {
```

Comment:
I think most of these methods should be included in the Decorator class in Dekorate, but probably `withDeploymentContext`.

Reply:
We could possilby have the `getMessage` method. I don't see `isApplied` something that makes sense for general purpose solution for a lot of reasons (e.g. its shouldn't be the devs responsibility to implement its logic).